### ISSUE TITLE
Listener keystore configuration doesn't work

### ISSUE THREAD

**Issue by praminda** [ballerina-platform/ballerina-lang#17600]
**Description:**
In WSO2 MGW we have three endpoints/ports listening for traffic.
- 9090 HTTP Gateway communication
- 9095 HTTPS Gateway Communication
- 9096 HTTPS Token services

Inside microgw config file which is provided as a ballerina config, we define the default keystore configurations and load listener level keystore configurations from this file. However if try to provide per listener keystore configurations through code ballerina throws below error.

```
error: {ballerina/http}HTTPError {message:"Conflicting configuration detected for listener configuration id 0.0.0.0:9096"}
	at ballerina/http:initEndpoint(<native>)
	   ballerina/http:init(service_endpoint.bal:85)
	   ballerina/http:__init(service_endpoint.bal:50)
```
Definition of a listener.

```java
listener http:Listener tokenListenerEndpoint = new (
    
        gateway:getConfigIntValue(gateway:LISTENER_CONF_INSTANCE_ID, gateway:TOKEN_LISTENER_PORT, 9096), config = {
    
        host: gateway:getConfigValue(gateway:LISTENER_CONF_INSTANCE_ID, gateway:LISTENER_CONF_HOST, "localhost"),
        secureSocket: {
            keyStore: {
                path: gateway:getConfigValue(gateway:LISTENER_CONF_INSTANCE_ID, gateway:LISTENER_CONF_KEY_STORE_PATH,
                    "${ballerina.home}/bre/security/ballerinaKeystore.p12"),
                password: gateway:getConfigValue(gateway:LISTENER_CONF_INSTANCE_ID,
                    gateway:LISTENER_CONF_KEY_STORE_PASSWORD, "xxxxx")
            }
        }
    }
);
```

**Steps to reproduce:**

**Affected Versions:**
0.990.5

**OS, DB, other environment details and versions:**
Any

**Related Issues (optional):**

**Suggested Labels (optional):**

Priority/Highest

**Suggested Assignees (optional):**

**Comment by praminda** [comment]
After further debugging I noticed this error is coming due to port `9096` being initialized two times. However still all listeners are being configured with the keystore defined in the configuration file. Not the hard coded listenerConfiguration.
Sample MGW project source files with the error is attached.
[mgw.zip](https://github.com/ballerina-platform/ballerina-lang/files/3496050/mgw.zip)

**Comment by Bhashinee** [comment]
I tried re-creating the issue with 2 listeners. One listener getting the keystore from a config file and in the other listener I have hard coded the keystore config. But the above mentioned issue is not occurring in the 990.5 released version or in the 1.0.0 alpha release. 

Code sample is attached here,

**bal file**
```
import ballerina/http;
import ballerina/log;
import ballerina/config;

http:ServiceEndpointConfiguration helloWorldEPConfig = {
    secureSocket: {
        keyStore: {
            path: config:getAsString("keystore.file"),
            password: config:getAsString("keystore.password")
        }
    }
};

http:ServiceEndpointConfiguration helloWorldEP2Config = {
    secureSocket: {
        keyStore: {
            path: "${ballerina.home}/bre/security/ballerinaKeystore.p12",
            password: "ballerina"
        }
    }
};

listener http:Listener helloWorldEP = new(9096, config = helloWorldEPConfig);
listener http:Listener helloWorldEP2 = new(9095, config = helloWorldEP2Config);

@http:ServiceConfig {
    basePath: "/hello"
}
service helloWorld on helloWorldEP2 {
    @http:ResourceConfig {
        methods: ["GET"],
        path: "/"
    }
    resource function sayHello(http:Caller caller, http:Request req) {
        var result = caller->respond("Successful");
        if (result is error) {
            log:printError("Error in responding", err = result);
        }
    }
}

@http:ServiceConfig {
    basePath: "/hello"
}
service helloWorld2 on helloWorldEP {
    @http:ResourceConfig {
        methods: ["GET"],
        path: "/"
    }
    resource function sayHello2(http:Caller caller, http:Request req) {
        var result = caller->respond("Successful");
        if (result is error) {
            log:printError("Error in responding", err = result);
        }
    }
}
```
**conf file**
```
[keystore]
file="/home/bhashinee/TiggerStack/finalCertificates/localcrt.p12"
password="localpwd"
```
If this is not the scenario that you are trying out, please give a simple reproducible code segment.

**Comment by Bhashinee** [comment]
Closing the issue as this is not reproducible from our end. Please re-open, if you have any update.
